config/steps/cachetool.yml

```diff
@@ -1,21 +1,42 @@
+- name: LEPHARE - Set cachetool facts based on PHP version
+  set_fact:
+    cachetool_facts:
+      version: |-
+        {{
+          'latest' if php_facts.version is version('8.0', '>=') else
+          '7.1.0'  if php_facts.version is version('7.3', '>=') else
+          '5.1.3'  if php_facts.version is version('7.2', '>=') else
+          '4.1.1'  if php_facts.version is version('7.1', '>=') else
+          '3.2.2'
+        }}
+      options: |-
+        {{
+          '--web=SymfonyHttpClient' if php_facts.version is version('7.3', '>=') else
+          '--web'
+        }}
+
 - name: LEPHARE - Check for cachetool.phar
   stat: path={{ lephare_cachetool_path }}
   register: cachetool_file
 
 - name: LEPHARE - Run cachetool self-update
+  ignore_errors: true
   shell: "{{ lephare_cachetool_path }} self-update"
-  when: cachetool_file.stat.exists and lephare_cachetool_self_update
-  ignore_errors: True
+  when: cachetool_file.stat.exists and lephare_cachetool_self_update and cachetool_facts.version == 'latest'
```

Comment:
why `cachetool_facts.version == 'latest'` ? 
a cachetool 5.1.3 should upgrade to 5.1.4 if available ?

Reply:
@pierreboissinot for the record, cachetool does not check php requirements when self-update is called. This is why self-update task is called only if the `latest` version is expected